nighthawk/include/nighthawk/common/statistic.h

```diff
@@ -63,9 +63,24 @@ class Statistic : Envoy::NonCopyable {
    * single global view. Types of the Statistics objects that will be combined
    * must be the same, or else a std::bad_cast exception will be raised.
    * @param statistic The Statistic that should be combined with this instance.
-   * @return std::unique_ptr<Statistic> instance.
+   * @return StatisticPtr instance.
    */
-  virtual std::unique_ptr<Statistic> combine(const Statistic& statistic) PURE;
+  virtual std::unique_ptr<Statistic> combine(const Statistic& statistic) const PURE;
```

Comment:
Nit:pProbably could define the `StatisticPtr` above using forward decl of `Statistic`.